python/modules/variable.py

```diff
@@ -0,0 +1,381 @@
+    """
+    Parameters with type 'part' are represented in  xml settings
+    as following:
+    <parameter key = <KEY> value = <VALUE> type = <TYPE> part = <PART>/>
+    
+    i.e.
+    <parameter key = "input_path" value = "C://" type = "path" part = "1"/>
+    <parameter key = "input_path" value = "filename_1" type = "ref" part = "2"/>
+    """
+    def __init__(self, key, value, args = {}):
+        Variable.__init__(self, key, value, args)
+
+    def _check_paths_presence(self, var_dict, env):
+        """
+        Verifies if there is any path in Variable's parts.
+        """
+        for key, variable in var_dict.items():
+            var = variable.get_variable(env)
+            if isinstance(var, VariablePath):
+                return True
+        return False
+
+    def get_value(self, env):
+        return self._get_converted_value(env)[0]
+
+    def get_variable(self, env):
+        # creates variable with merged value
+        merged_value, contains_path = self._get_converted_value(env)
+        if contains_path:
+            var = VariablePath(self.key, merged_value)
+        else:
+            var = Variable(self.key, merged_value)
+        return var
+
+    def _get_converted_value(self, env, args = {}):
+        """
+        Gets converted value (depending on paths presence).
+        Returns converted value and boolean 
+        (contains_path = True/False).
+        """
+        order = sorted(self.value.keys())
+        values_list = []
+        for key in order:
+            v_part = self.value[key].get_value(env)
+            values_list.append(str(v_part))
+        contains_path = self._check_paths_presence(self.value, env)
+        if contains_path:
+            merged_value = FM.path_join(*values_list)
+        else:
+            merged_value = "".join(values_list)  
+        logger.debug("Merged variable %s: %s", self.key, merged_value)
+        return merged_value, contains_path
+
+class VariableList(Variable):
+    """
+    Parameters with type 'list' are represented in  xml settings
+    as following:
+    <parameter key = <KEY> type = "list">
+        <parameter value = <VALUE> type = <TYPE> />
+        <parameter value = <VALUE> type = <TYPE> />
+    </parameter>
+    
+    i.e.
+    <parameter key = "csv_data" type = "list">
+        <parameter value = "Nuclei.csv">
+        <parameter value = "ShrinkedNuclei.csv">
+    </parameter>
+    """
+    def __init__(self, key, value, args = {}):
+        Variable.__init__(self, key, value, args)
+
+    def get_variable(self, env):
+        converted_list = self.get_value(env)
+        variable = VariableList(self.key, converted_list)
+        return variable
+
+    def get_value(self, env):
+        converted_list = []
+        raw_list = self.value
+        for var in raw_list:
+            new_var = var.get_variable(env)
+            converted_list.append(new_var)
+        return converted_list
+        
+    def get_converted_value(self, env):
```

Comment:
Well, yes, but in VariableParted it is called anyway by get_value() function so there is no need to leave it as global function.

Function get_converted_value of VariableList and VariableStructure is no longer used in other modules, according to few major changes in variable module, but i would like to leave them for a moment in case of future needs. I can comment them by now.